activerecord/lib/active_record/connection_adapters/sqlite3/database_statements.rb

```diff
@@ -43,7 +43,7 @@ def exec_query(sql, name = nil, binds = [], prepare: false)
 
           type_casted_binds = type_casted_binds(binds)
 
-          log(sql, name, binds, type_casted_binds) do
+          log(sql, name, binds, type_casted_binds, nil, background) do
```

Comment:
Now that I wrote the PR description, I think we shouldn't call `log` for background queries, but store the necessary data to then call it back in the main thread inside `FutureResult#result!`.